misc/discv5/src/config.rs

```diff
@@ -7,6 +7,11 @@ pub struct Discv5Config {
     /// The request timeout for each UDP request. Default: 4 seconds.
     pub request_timeout: Duration,
 
+    /// The query timeout after which a `QueryPeer` is marked unresponsive.
+    /// Unresponsive peers don't count towards the parallelism limits for a query.
+    /// Hence, we may potentially end up making more requests to good peers.
+    pub query_timeout: Duration,
```

Comment:
We should add a function to the Discv5ConfigBuilder to allow configuration of this new parameter